CONTRIBUTING.md

```diff
@@ -1,51 +1,147 @@
-## Local development
+## Table of Contents
 
-### Pre-requisites
+- [Setting Up Local Development Environment](#setting-up-local-development-environment)
+  - [Prerequisites for Development](#prerequisites-for-development)
+  - [Building, Running, and Testing the Repository](#building-running-and-testing-the-repository)
+  - [Setting Up Local Language Servers](#setting-up-local-language-servers)
+    - [Roslyn](#roslyn)
+    - [Razor](#razor)
+  - [Configuring Local Language Servers](#configuring-local-language-servers)
+    - [Roslyn](#roslyn-1)
+    - [Razor](#razor-1)
+  - [Debugging Local Language Servers](#debugging-local-language-servers)
+- [Creating VSIX Packages for the Extension](#creating-vsix-packages-for-the-extension)
+- [Updating the `Roslyn` Language Server Version](#updating-the-roslyn-language-server-version)
 
-First install:
+## Setting Up Local Development Environment
+
+Setting up your local development environment for the vscode-csharp repository involves several steps. This guide will walk you through the process.
+
+### Prerequisites for Development
+
+Before you start, make sure you have the following software installed on your machine:
 
 * Node.js ([v18.17.0 LTS](https://nodejs.org/en/blog/release/v18.17.0) is recommended).
 * Npm (The version shipped with node is fine)
-* .NET 7.0 SDK (dotnet should be on your path)
+* .NET 8.0 SDK (dotnet should be on your path)
+
+Once you have these installed, you can proceed to clone the vscode-csharp repository:
+
+```bash
+git clone https://github.com/your-username/vscode-csharp.git
+cd vscode-csharp
+```
+
+Next, install the necessary npm packages:
+
+```bash
+npm install
+```
+
+Now, you're ready to start building, running, and testing the repository.
+
+### Building, Running, and Testing the Repository
+
+Follow these steps to build, run, and test the repository:
+
+#### Building
+
+1. Run `npm i` - This command installs the project dependencies.
+2. Run `npm i -g gulp` - This command installs Gulp globally.
+3. Run `gulp installDependencies` - This command downloads the various dependencies as specified by the version in the [package.json](package.json) file.
+4. Run `code .` - This command opens the project in Visual Studio Code.
+5. Run `npm run watch` (Optional) - This command watches for code changes.
+
+#### Running
+
+After completing the build steps:
+
+1. Press <kbd>Ctrl+Shift+D</kbd> to open the Run view in VS Code and ensure `Launch Extension` is selected.
+2. Start debugging by pressing <kbd>F5</kbd>.
+
+#### Testing
+
+To run tests:
+
+1. Execute `npm run test` or press <kbd>F5</kbd> in VS Code with the "Launch Tests" debug configuration selected.
+2. For integration tests, select `Razor Integration Tests` and press <kbd>F5</kbd> to start debugging.
```

Comment:
For integration tests it might be better to mention the two 'current file' integration tests (for roslyn and for razor). those will allow you to actually debug the test, the 'Razor integration tests' configuration does not.